File: Python/Problem_059.py
def read_cipher(path: str) -> list[int]:
    with open(path, "r", encoding="utf-8") as f:
        return [int(x) for x in f.read().strip().split(",")]

def is_printable_ascii(x: int) -> bool:
    # allow newline (10) and standard printable range
    return x == 10 or 32 <= x <= 126

def score_english(text: str) -> int:
    # crude but effective for this dataset
    common = [" the ", " and ", " of ", " to ", " in ", " that ", " is ", " was "]
    t = " " + text.lower() + " "
    return sum(t.count(w) for w in common)

def euler_59(path="0059_cipher.txt") -> tuple[str, int]:
    cipher = read_cipher(path)

    best_score = -1
    best_key = ""
    best_plain = ""

    for a in range(ord('a'), ord('z') + 1):
        for b in range(ord('a'), ord('z') + 1):
            for c in range(ord('a'), ord('z') + 1):
                key = [a, b, c]
                plain = []
                ok = True

                for i, val in enumerate(cipher):
                    p = val ^ key[i % 3]
                    if not is_printable_ascii(p):
                        ok = False
                        break
                    plain.append(p)

                if not ok:
                    continue

                text = "".join(chr(x) for x in plain)
                s = score_english(text)

                if s > best_score:
                    best_score = s
                    best_key = "".join(chr(x) for x in key)
                    best_plain = text

    ascii_sum = sum(ord(ch) for ch in best_plain)
    return best_key, ascii_sum

if __name__ == "__main__":
    key, total = euler_59()
    print(key, total)  # god 129448 (with official file)

File: Java/Problem_059.java
import java.io.*;
import java.util.*;

public class Euler59 {

    static int[] readCipher(String path) throws IOException {
        String s;
        try (BufferedReader br = new BufferedReader(new FileReader(path))) {
            s = br.readLine();
        }
        String[] parts = s.split(",");
        int[] arr = new int[parts.length];
        for (int i = 0; i < parts.length; i++) arr[i] = Integer.parseInt(parts[i].trim());
        return arr;
    }

    static boolean printable(int p) {
        return p == 10 || (p >= 32 && p <= 126);
    }

    static int scoreEnglish(String text) {
        String t = " " + text.toLowerCase() + " ";
        String[] common = {" the ", " and ", " of ", " to ", " in ", " that ", " is ", " was "};
        int score = 0;
        for (String w : common) {
            int idx = 0;
            while ((idx = t.indexOf(w, idx)) != -1) {
                score++;
                idx += 1;
            }
        }
        return score;
    }

    public static void main(String[] args) throws Exception {
        int[] cipher = readCipher("0059_cipher.txt");

        int bestScore = -1;
        String bestKey = "";
        String bestPlain = "";

        for (int a = 'a'; a <= 'z'; a++) {
            for (int b = 'a'; b <= 'z'; b++) {
                for (int c = 'a'; c <= 'z'; c++) {
                    int[] key = {a, b, c};
                    StringBuilder sb = new StringBuilder(cipher.length);
                    boolean ok = true;

                    for (int i = 0; i < cipher.length; i++) {
                        int p = cipher[i] ^ key[i % 3];
                        if (!printable(p)) { ok = false; break; }
                        sb.append((char)p);
                    }
                    if (!ok) continue;

                    String plain = sb.toString();
                    int sc = scoreEnglish(plain);
                    if (sc > bestScore) {
                        bestScore = sc;
                        bestKey = "" + (char)a + (char)b + (char)c;
                        bestPlain = plain;
                    }
                }
            }
        }

        long sum = 0;
        for (int i = 0; i < bestPlain.length(); i++) sum += (int)bestPlain.charAt(i);

        System.out.println(bestKey + " " + sum); // god 129448 (official file)
    }
}
